gooddata-java-model/src/main/java/com/gooddata/sdk/model/settings/SettingItem.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright (C) 2007-2021, GoodData(R) Corporation. All rights reserved.
+ */
+package com.gooddata.sdk.model.settings;
+
+import com.fasterxml.jackson.annotation.*;
+import com.gooddata.sdk.common.util.GoodDataToStringBuilder;
+
+import static com.gooddata.sdk.common.util.Validate.notEmpty;
+import static com.gooddata.sdk.common.util.Validate.notNullState;
+
+@JsonTypeName("settingItem")
+@JsonTypeInfo(include = JsonTypeInfo.As.WRAPPER_OBJECT, use = JsonTypeInfo.Id.NAME)
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+public class SettingItem {
+
+    private final String name;
+    private final String source;
+    private String enabled;
+
+    @JsonIgnore
+    private Links links;
+
+    /**
+     * Creates new setting item with given value and default source (catalog).
+     *
+     * @param name unique name of setting item
+     * @param enabled true (setting item enabled) or false (setting item disabled)
+     * @param source source of settings
+     */
+    public SettingItem(String name, String enabled, String source) {
+        this(notEmpty(name, "name"), notEmpty(enabled, "enabled"),
+                notEmpty(source, "source"), null
+        );
+    }
+
+    @JsonCreator
+    SettingItem(@JsonProperty("key") String name, @JsonProperty("value") String enabled,
+                @JsonProperty("source") String source, @JsonProperty("links") Links links) {
+        this.name = name;
+        this.enabled = enabled;
+        this.source = source;
+        this.links = links;
+    }
+
+    @JsonProperty("key")
+    public String getName() {
+        return name;
+    }
+
+    @JsonProperty("value")
+    public String isEnabled() {
+        return enabled;
+    }
+
+    public String getSource() {
+        return source;
+    }
+
+    public Links getLinks() {
+        return links;
+    }
+
+    public void setEnabled(String enabled) {
+        this.enabled = enabled;
+    }
+
+    public void setUri(String uri) {
```

Comment:
Is this setter necessary? I don't know about use case where would normal user need to change the URI.

Reply:
I used it for testing to avoid to using reflection in that case, but it probably won't matter in that case. Okay, I'll refactored it.

Reply:
Removed